### PR TITLE
Add organization name & logo to the page API response

### PR DESCRIPTION
**Pull request by JoeyStk**
### Short description
This PR adds the organization name & logo to the page API response. It is the first time I'm really working with the API so feel free to give detailed tips/background information on how we handle things with our API.

### Proposed changes

- Add organization name & logo to page API response

### Side effects

- I think none

### Resolved issues

Fixes:  #1322

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

**Review by timoludwig — CHANGES_REQUESTED** [review]

**Review by timoludwig — CHANGES_REQUESTED** [review]
Thanks a lot! :+1:

The API response looks good.

> ### Side effects
> 
>     * I think none

I think I found a side effect: For each page which has an organizaion, the API makes one additional db query:

Screenshot 2023-02-21 at 12-52-20 Screenshot

So when there are many pages with organizations, this will slow the API response down.
Could you fetch the organization data together with the pages queryset (see [`select_related()`](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#django.db.models.query.QuerySet.select_related))?

Also, I think it makes sense to add a few pages to the test organization in the test data and adapt the expected API responses.

File: integreat_cms/api/v3/pages.py [hunk 1]
```diff
@@ -76,7 +76,9 @@ def transform_page(page_translation):
         "parent": parent,
         "order": order,
         "available_languages": page_translation.available_languages_dict,
-        "thumbnail": page_translation.page.icon.url
+        "thumbnail": page_translation.page.icon.url,
+        "organisation_name": page_translation.organization.name,
+        "organisation_icon": page_translation.organization.icon.url
         if page_translation.page.icon
         else None,
```
[the remark this answers is not in the document]
Comment (timoludwig) [reply integreat_cms/api/v3/pages.py]: Point 3. was not addressed

**Comment by membralala** [comment]
Hey @Timo, thank you for your detailed feedback. Johannes and I talked about the prefetching and we are both still confused by it so I am afraid I have to ask you for advice once again 👀. 
As far as I understand this we want to use the `select_related` method, where `transform_page` function is called in a loop(?), however, what I'm confused about is: How do those two comments relate? 
1. > So when there are many pages with organizations, this will slow the API response down.
Could you fetch the organization data together with the pages queryset (see [select_related()](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#django.db.models.query.QuerySet.select_related))?

2. > Please prefetch the organization objects from the database to make sure we don't decrease the API performance significantly by this feature.

Is 1. just a more detailed instruction to 2. or are those two separate steps that should be done to improve performance? 

Also, how do you configure your django toolbar to show debug information for queries caused by api calls? 🙈 
Thanks in advance!

**Comment by timoludwig** [comment]
> Is 1. just a more detailed instruction to 2. or are those two separate steps that should be done to improve performance?

It's 1., I meant the same thing with both comments. At the moment, the organization data is queried for each page separately in the loop, causing one db query per page with organization. Instead, Django provides the helper [`select_related()`](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#django.db.models.query.QuerySet.select_related), which allows us to query the organizations for all pages at once (Django translates this to a `INNER JOIN` in SQL). This has to be done outside of the loop in order to work, so every time we retrieve the queryset of pages, we have to select the organization data via `.select_related("organization")` as well, e.g. here:
- https://github.com/digitalfabrik/integreat-cms/blob/ad9fd94c7f6d80850bf9027980cbaa13fc4a1d6f/integreat_cms/api/v3/pages.py#L109
- https://github.com/digitalfabrik/integreat-cms/blob/ad9fd94c7f6d80850bf9027980cbaa13fc4a1d6f/integreat_cms/api/v3/pages.py#L151
- https://github.com/digitalfabrik/integreat-cms/blob/ad9fd94c7f6d80850bf9027980cbaa13fc4a1d6f/integreat_cms/api/v3/pages.py#L248
- https://github.com/digitalfabrik/integreat-cms/blob/ad9fd94c7f6d80850bf9027980cbaa13fc4a1d6f/integreat_cms/api/v3/pages.py#L309

> Also, how do you configure your django toolbar to show debug information for queries caused by api calls? :see_no_evil: Thanks in advance!

See the documentation: https://digitalfabrik.github.io/integreat-cms/debugging.html

**Comment by membralala** [comment]
@timoludwig Thank you very much for your detailed explanation! With that we added the prefetch-mechanism.

**Review by timoludwig — APPROVED** [review]
Thanks a lot! :+1:
Yes, this solves the performance issues! :rocket:

**Review by MizukiTemma — APPROVED** [review]
Thank you 👍 Looks good!

**Pull request merged**